improver_tests/cli/test_init.py

```diff
@@ -206,6 +206,17 @@ def test_basic(self, m):
         m.assert_any_call(load_cube, "foo", constraints='wind_from_direction')
         self.assertEqual(m.call_count, 2)
 
+    @patch('improver.cli.maybe_coerce_with', return_value='return')
+    def test_list(self, m):
+        """Tests that a list returns a function which itself returns 2 cubes"""
+        result = create_constrained_inputcubelist_converter(
+            ['wind_speed'], ['wind_from_direction'])
```

Comment:
Can we try a few more versions of this test, please, namely:

- A test with two / multiple items in each list, only one of which is valid (passes)
- A test with two / multiple items in each list where more than one is valid (should fail, I think?)
- Something that demonstrates what the behaviour is if the lists are of mismatched lengths (eg 2 possible names for one cube but only one for the other)

Reply:
Yes, no problem. I've added some more tests to check this behaviour.

Reply:
This looks clever and potentially dirty.  However this much improves the test coverage, and since we have a ticket on the board to deal with any technical issues with the implementation, I'm happy to approve this and refactor later if need be.